### PR TITLE
Fix uninitialized array attributes

### PR DESCRIPTION
**Pull request by sklam**
... after slicing and other view creating operations.

Also, enhanced the contiguous test for CUDA.

### CODE REVIEW

**Comment by gmarkall** [comment]
Could the construction of the array struct for nested arrays (around line 460 of numba/targets/base.py) also be changed to call `populate_array`?

**Comment by sklam** [comment]
Yes, and it should.

File: numba/targets/builtins.py [hunk 1]
```diff
@@ -1420,6 +1420,13 @@ def array_ravel_impl(context, builder, sig, args):
     flatarr.shape = cgutils.pack_array(builder, [size])
```
Comment (gmarkall) [inline numba/targets/builtins.py]: Are these assignments now redundant?

Comment (sklam) [inline numba/targets/builtins.py]: oops

**Comment by gmarkall** [comment]
This looks good now.

**Pull request merged**